core/controllers/classifier_test.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the controllers that communicate with VM for training
+classifiers."""
+
+import hashlib
+import hmac
+import json
+import os
+
+from core.domain import classifier_services
+from core.domain import exp_services
+from core.tests import test_utils
+import feconf
+
+
+def generate_signature(data):
```

Comment:
If you pull out the function I mentioned above (in the review of the controller file), you could use it here, and not need to create a duplicate function in the tests.